gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/spi/PublisherApi.java

```diff
@@ -0,0 +1,508 @@
+        public Object extractNextToken(ListTopicsResponse payload) {
+          return payload.getNextPageToken();
+        }
+        @Override
+        public Iterable<Topic> extractResources(ListTopicsResponse payload) {
+          return payload.getTopicsList();
+        }
+      };
+
+  private static PageDescriptor<ListTopicSubscriptionsRequest, ListTopicSubscriptionsResponse, String> LIST_TOPIC_SUBSCRIPTIONS_PAGE_DESC =
+      new PageDescriptor<ListTopicSubscriptionsRequest, ListTopicSubscriptionsResponse, String>() {
+        @Override
+        public Object emptyToken() {
+          return "";
+        }
+        @Override
+        public ListTopicSubscriptionsRequest injectToken(
+            ListTopicSubscriptionsRequest payload, Object token) {
+          return ListTopicSubscriptionsRequest
+            .newBuilder(payload)
+            .setPageToken((String) token)
+            .build();
+        }
+        @Override
+        public Object extractNextToken(ListTopicSubscriptionsResponse payload) {
+          return payload.getNextPageToken();
+        }
+        @Override
+        public Iterable<String> extractResources(ListTopicSubscriptionsResponse payload) {
+          return payload.getSubscriptionsList();
+        }
+      };
+
+  private static String ALL_SCOPES[] = {
+    "https://www.googleapis.com/auth/pubsub"
+  };
+
+  /**
+   * A PathTemplate representing the fully-qualified path to represent
+   * a project resource.
+   */
+  public static final PathTemplate PROJECT_PATH_TEMPLATE =
+      PathTemplate.create("/projects/{project}");
+
+  /**
+   * A PathTemplate representing the fully-qualified path to represent
+   * a topic resource.
+   */
+  public static final PathTemplate TOPIC_PATH_TEMPLATE =
+      PathTemplate.create("/projects/{project}/topics/{topic}");
+
+  // ========
+  // Members
+  // ========
+
+  private final ManagedChannel channel;
+  private final ServiceApiSettings settings;
+
+  // ===============
+  // Factory Methods
+  // ===============
+
+  /**
+   * Constructs an instance of PublisherApi with default settings.
+   */
+  public static PublisherApi create() throws IOException {
+    return create(new ServiceApiSettings());
+  }
+
+  /**
+   * Constructs an instance of PublisherApi, using the given settings. The channels are created based
+   * on the settings passed in, or defaults for any settings that are not set.
+   */
+  public static PublisherApi create(ServiceApiSettings settings) throws IOException {
+    return new PublisherApi(settings);
+  }
+
+  /**
+   * Constructs an instance of PublisherApi, using the given settings. This is protected so that it
+   * easy to make a subclass, but otherwise, the static factory methods should be preferred.
+   */
+  protected PublisherApi(ServiceApiSettings settings) throws IOException {
+    ServiceApiSettings internalSettings = ApiUtils.settingsWithChannels(settings,
+        SERVICE_ADDRESS, DEFAULT_SERVICE_PORT, ALL_SCOPES);
```

Comment:
1. null channel, host and port given: yes, your host and port will be used. 

https://github.com/garrettjonesgoogle/gcloud-java/blob/pubsub-alpha/gcloud-java-gax/src/main/java/io/gapi/gax/internal/ApiUtils.java#L94

https://github.com/garrettjonesgoogle/gcloud-java/blob/pubsub-alpha/gcloud-java-gax/src/main/java/io/gapi/gax/internal/ApiUtils.java#L99

2. null channel, credentials is provided: it looks like the credentials won't be used. This appears to be a mistake - I will fix it.